### PR TITLE
MRG: remove call to calc_lwork

### PR DESCRIPTION
**Pull request by bthirion**
addresses #353

### CODE REVIEW

File: nipy/algorithms/statistics/utils.py [hunk 1]
```diff
@@ -38,37 +38,21 @@ def multiple_fast_inv(a):
     This function is borrowed from scipy.linalg.inv, 
     but with some customizations for speed-up.
     """
+    # from scipy.linalg.decomp import _asarray_validated
```
Comment (matthew-brett) [inline nipy/algorithms/statistics/utils.py]: Is this an old debug comment?  Remove?

File: nipy/algorithms/statistics/utils.py [hunk 2]
```diff
@@ -38,37 +38,21 @@ def multiple_fast_inv(a):
...
     if a.shape[1] != a.shape[2]:
         raise ValueError('a must have shape(n_samples, n_dim, n_dim)')
-    from scipy.linalg import calc_lwork
-    from scipy.linalg.lapack import  get_lapack_funcs
-    a1, n = a[0], a.shape[0]
-    getrf, getri = get_lapack_funcs(('getrf', 'getri'), (a1,))
-    for i in range(n):
-        if (getrf.module_name[:7] == 'clapack'
-            and getri.module_name[:7] != 'clapack'):
-            # ATLAS 3.2.1 has getrf but not getri.
```
Comment (matthew-brett) [inline nipy/algorithms/statistics/utils.py]: Is this still a problem - that some lapacks don't have this routine?

File: nipy/algorithms/statistics/utils.py (hunk 1 above)
Comment (bthirion) [inline nipy/algorithms/statistics/utils.py]: Yes.

File: nipy/algorithms/statistics/utils.py (hunk 2 above)
Comment (bthirion) [inline nipy/algorithms/statistics/utils.py]: I simply don't know.

Comment (matthew-brett) [inline nipy/algorithms/statistics/utils.py]: I don't really understand the `clapack` `flapack` thing.  Does it matter which one we find?  I notice that the Debian builds I have fine `flapack` versions of these for scipy back to 0.7.0.

File: nipy/algorithms/statistics/utils.py [hunk 3]
```diff
@@ -38,37 +38,20 @@ def multiple_fast_inv(a):
...
-            a[i] = np.transpose(lu)
-        else:
-            a[i], piv, info = getrf(a[i], overwrite_a=True)
+    a = np.asarray_chkfinite(a)
+    getrf, getri = get_lapack_funcs(('getrf', 'getri'), (a[0],))
+    for i in range(a.shape[0]):
+        lu, piv, info = getrf(a[i], overwrite_a=True)
         if info == 0:
-            if getri.module_name[:7] == 'flapack':
-                lwork = calc_lwork.getri(getri.prefix, a1.shape[0])
```
Comment (matthew-brett) [inline nipy/algorithms/statistics/utils.py]: Does it matter that we leave `lwork` at its default by removing this?

**Comment by matthew-brett** [comment]
I had a go at porting the most recent scipy code, along with fixes for older scipies, in a PR to your branch here:

https://github.com/bthirion/nipy/pull/3

I've tested it with scipy 0.10 through 0.16, so I think it is right.

**Comment by matthew-brett** [comment]
I propose to merge this one with my patch soon - any objections?

**Comment by bthirion** [comment]
I'd be happy to merge, but there is a travis failure (which seems related to coverage issues ??)

**Comment by matthew-brett** [comment]
Yes, the travis failure is unrelated.

**Comment by bthirion** [comment]
That should be fine.

**Comment by matthew-brett** [comment]
Thanks too - merging.

**Pull request merged**